### PR TITLE
Only get Jetpack Sites for Jetpack Cloud

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Change `me/sites` endpoint to only request Jetpack sites for Jetpack Cloud

#### Testing instructions

1. Navigate to `/backups`
2. Verify the list of sites is only jetpack and atomic sites
3. Boot Calypso
4. Verify the sites visible are the same as on WordPress.com

#### Next Steps

* Going Directly to a WPCOM site via the site slug ( `/backups/:wpcomSiteSlug`  ) will load up that site. Should we filter the `/sites` endpoint as well?
* The site selector site count is pulled from user info. We need to either provide fields on user info or replace the count.
Screen Shot 2020-04-01 at 10 42 13 AM

### CODE REVIEW

File: client/state/sites/actions.js [hunk 1]
```diff
@@ -90,6 +91,7 @@ export function requestSites() {
 				site_activity: 'active',
 				fields: SITE_REQUEST_FIELDS,
 				options: SITE_REQUEST_OPTIONS,
+				filters: config.isEnabled( 'jetpack-cloud' ) ? 'atomic,jetpack' : '',
```
Comment (enejb) [inline client/state/sites/actions.js:94]: Do we want atomic here?
My answer right now is no.

**Comment by enejb** [comment]
> Going Directly to a WPCOM site via the site slug ( `/backups/:wpcomSiteSlug` ) will load up that site. Should we filter the /sites endpoint as well?

Yeah I think adding a filter to the /sites endpoint would be helpful here as well.

File: client/state/sites/actions.js (hunk 1 above)
Comment (enejb) [inline client/state/sites/actions.js:94]: Instead of checking if jetpack-cloud is enabled why not pass in the sites_filter config value?

**Comment by enejb** [comment]
To be addressed in future PRs 
I think we should make sure that 
Screen Shot 2020-04-03 at 12 20 35 PM

Also makes sense. Right now it shows count all the users sites.

File: client/state/sites/actions.js (hunk 1 above)
Comment (keoshi) [reply client/state/sites/actions.js:94]: I would agree. For all intents and purposes, Atomic sites are dotcom sites in terms of UX so we should do our best to not fork their experience. That means avoiding bringing them into the Cloud but also not list Atomic sites inside JP Cloud, as I think is what's happening here.

Comment (cbauerman) [reply client/state/sites/actions.js:94]: yeah, that seems like a more easily configurable approach!

**Pull request merged**